incidents/src/main/java/fr/openent/incidents/helper/PunishmentHelper.java

```diff
@@ -385,44 +371,39 @@ private JsonObject getCursor() {
         return new JsonObject().put("batchSize", 2147483647);
     }
 
-    private void mapGetterResults(JsonArray result, Handler<AsyncResult<JsonArray>> handler) {
+    @SuppressWarnings("unchecked")
+    private void mapGetterResults(JsonArray result, String order, boolean reverse, Handler<AsyncResult<JsonArray>> handler) {
         if (result.size() == 0) {
             handler.handle(Future.succeededFuture(result));
             return;
         }
 
         List<String> ownerIds = ((List<JsonObject>) result.getList())
-                .stream()
-                .map(res -> res.getString("owner_id"))
-                .collect(Collectors.toList());
+                .stream().map(res -> res.getString(Field.OWNER_ID)).collect(Collectors.toList());
 
         List<String> studentIds = ((List<JsonObject>) result.getList())
-                .stream()
-                .map(res -> res.getString("student_id"))
-                .collect(Collectors.toList());
+                .stream().map(res -> res.getString(Field.STUDENT_ID)).collect(Collectors.toList());
 
         List<Long> typeIds = ((List<JsonObject>) result.getList())
-                .stream()
-                .map(res -> res.getLong("type_id"))
-                .collect(Collectors.toList());
+                .stream().map(res -> res.getLong(Field.TYPE_ID)).collect(Collectors.toList());
 
-        Future<JsonArray> ownersFuture = Future.future();
-        Future<JsonArray> studentsFuture = Future.future();
-        Future<JsonArray> typesFuture = Future.future();
+        Promise<JsonArray> ownersPromise = Promise.promise();
+        Promise<JsonArray> studentsPromise = Promise.promise();
+        Promise<JsonArray> typesPromise = Promise.promise();
 
         userService.getUsers(ownerIds, resUsers -> {
             if (resUsers.isLeft()) {
-                ownersFuture.fail("[Incidents@Punishment::mapGetterResults] Failed to get owners");
+                ownersPromise.fail("[Incidents@PunishmentHelper::mapGetterResults] Failed to get owners " + resUsers.left().getValue());
```

Comment:
Je ne sais pas si c'est une bonne pratique de remonter l'erreur dans la réponse. Ce serait mieux si c'était porté par un log plutôt, non ?

Reply:
Je suis d'accord, tu peux mettre un log d'erreur avec puis mettre seulement dans la promise.fail resUsers.left().getValue();

Reply:
Je voulais dire l'inverse 
log.error(ownersPromise.fail("[Incidents@PunishmentHelper::mapGetterResults] Failed to get owners " + resUsers.left().getValue());
ownersPromise.fail("[Incidents@PunishmentHelper::mapGetterResults] Failed to get owners");